docs(BTTv2): Added documention about tests report (#45)
bonitasoft/bonita-test-toolkit-doc: modules/ROOT/pages/tests-report.adoc

= Tests report

> :description: This pages describes the report generated once the tests are run.
> 
> Understand the report generated by Bonita Test Toolkit when the tests are run.
> 

Reviewer: ```suggestion
:description: This page provides an explanation of the report generated after running the tests using Bonita Test Toolkit.

Learn about the insights and information presented in the report here.
```

== Purpose of the report

The main purpose of the report is to visualize the process execution,
thanks to the https://github.com/process-analytics/bpmn-visualization-js[bpmn-visualization] library.

This is realized by decorating the process diagrams with:

* Badges on the tasks, showing the number of `Executed`, `Ready/Waiting` or `Failed` tasks
* Highlight of the road followed by process instances (or cases) execution

Additionally, a set of metrics are provided for each test: execution time, duration,...

== How to generate the report

A report (html file) is generated once all the tests of a test file is run.
This report can be found in the `target/bonita-test-reports` directory of your tests.

You can de-activate the report generation by calling the `toolkit.deactivateTestReports()` method.
This won't generate a report for the whole test file (even if it is called in a single test).

=== Multiple process instances and sub-process

If several process instances are launched in a test, a diagram is displayed for each instance.
The process diagram shows a correlation of all the instances.
[NOTE]

> A maximum of 5 process instances are displayed, to avoid to pollute the report, in case of performance test for instance.

Reviewer: ```suggestion
To prevent report clutter during performance testing, only a maximum of 5 process instances will be displayed.
```
